jme3-effects/src/main/java/com/jme3/post/filters/ContrastAdjustmentFilter.java

```diff
@@ -0,0 +1,296 @@
+ *   notice, this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright
+ *   notice, this list of conditions and the following disclaimer in the
+ *   documentation and/or other materials provided with the distribution.
+ *
+ * * Neither the name of 'jMonkeyEngine' nor the names of its contributors
+ *   may be used to endorse or promote products derived from this software
+ *   without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jme3.post.filters;
+
+import com.jme3.asset.AssetManager;
+import com.jme3.export.InputCapsule;
+import com.jme3.export.JmeExporter;
+import com.jme3.export.JmeImporter;
+import com.jme3.export.OutputCapsule;
+import com.jme3.material.Material;
+import com.jme3.post.Filter;
+import com.jme3.renderer.RenderManager;
+import com.jme3.renderer.ViewPort;
+import java.io.IOException;
+
+/**
+ * A color filter used to adjust the contrast for color channels of the rendered scene by using a
+ * simple transfer function which involves adjusting the input range of the color channels based on
+ * an upper and a lower limit, adjusting the exponent of different color channels and scaling the output values
+ * before being processed by the fragment shader.
+ *
+ * @author pavl_g.
+ */
+public class ContrastAdjustmentFilter extends Filter {
+
+    protected float redChannelExponent;
+    protected float greenChannelExponent;
+    protected float blueChannelExponent;
+    //the lower value and the upper value of the inputRange
+    protected float lowerLimit;
+    protected float upperLimit;
+    //the final pass scale factor
+    protected float redChannelScale;
+    protected float greenChannelScale;
+    protected float blueChannelScale;
+    protected Material material;
+
+    /**
+     * Instantiates a default color contrast filter, default input range and default scale.
+     * Default values :
+     * - Exponents = 1.0f on all channels.
+     * - Input Range Lower Limit = 0f.
+     * - Input Range Upper Limit = 1f.
+     * - Scale = 1.0f on all channels.
+     */
+    public ContrastAdjustmentFilter() {
+        this(1f);
```

Comment:
The default constructor should set the filter `name`. Use the filters that already exist as a reference. 
Can you change it like this?

```java
public ContrastAdjustmentFilter() {
    super("ContrastAdjustmentFilter");
}
```

Reply:
Good points, thanks.

Reply:
Recent commits address these issues. Are they resolved?

Reply:
yes